### ISSUE TITLE
Sortable table header row buttons mis-aligned, missing `role` and incorrectly styled

### ISSUE THREAD

**Issue by jrbarnes9** [ONSdigital/design-system#1723]
Sortable table header row buttons mis-aligned, missing `role` and incorrectly styled

Header row buttons should be:
- inheriting link style `text-underline-position` 
- accessible with `role=link` attribute
- left-aligned with column cells content

![Screenshot 2021-10-05 at 16 02 38](https://user-images.githubusercontent.com/43346934/136050434-8c5a567c-a8cb-47ae-aada-7f786ac01e60.png)

_Originally posted by @jrbarnes9 in https://github.com/ONSdigital/design-system/discussions/102#discussioncomment-1430774_

**Comment by jrbarnes9** [comment]
Resolved in #1744
